aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllerSample.Tests/IntegrationTests/ApiIdeasControllerForSession.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNet.TestHost;
+using TestingControllersSample;
+using Xunit;
+using System.Linq;
+using System.Net.Http.Headers;
+
+namespace TestingControllerSample.Tests.IntegrationTests
+{
+    public class ApiIdeasControllerForSession
+    {
+        private readonly TestServer _server;
+        private readonly HttpClient _client;
+
+        public ApiIdeasControllerForSession()
+        {
+            _server = new TestServer(TestServer.CreateBuilder()
+                .UseEnvironment("Development")
+                .UseStartup<Startup>());
+            _client = _server.CreateClient();
+
+            // client always expects json results
+            _client.DefaultRequestHeaders.Clear();
+            _client.DefaultRequestHeaders.Accept.Add(
+                new MediaTypeWithQualityHeaderValue("application/json"));
+        }
+
+        [Fact]
+        public async Task ReturnsNotFoundForBadSessionId()
+        {
+            var response = await _client.GetAsync("/api/ideas/forsession/500");
+            Assert.Equal(HttpStatusCode.NotFound,response.StatusCode);
+        }
+
+        public class IdeaDto
```

Comment:
This is a great example of why the web site shouldn't use anonymous types. If it had real types, they could be used here.